From 6e423ff7394dfe7f01f9a10962847f7c3aeee783 Mon Sep 17 00:00:00 2001
From: Roger Christensen <rc@silicom.dk>
Date: Wed, 24 Aug 2022 14:23:18 +0200
Subject: [PATCH] spi: spi-altera-dfl: Add support for N5013 and N5014

The N5013 and N5014 cards are recognized by the Device Feature Header
revision number. Extend the spi-altera-dfl driver to recognize and
support these cards.

Signed-off-by: Roger Christensen <rc@silicom.dk>
---
 drivers/spi/spi-altera-dfl.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/spi/spi-altera-dfl.c b/drivers/spi/spi-altera-dfl.c
index 5d6e08c12dff..8b32956ac380 100644
--- a/drivers/spi/spi-altera-dfl.c
+++ b/drivers/spi/spi-altera-dfl.c
@@ -24,6 +24,7 @@
 
 #define FME_FEATURE_ID_MAX10_SPI	0xe
 #define FME_FEATURE_REV_MAX10_SPI_N5010	0x1
+#define FME_FEATURE_REV_MAX10_SPI_N5014	0x2
 
 #define SPI_CORE_PARAMETER      0x8
 #define SHIFT_MODE              BIT_ULL(1)
@@ -165,6 +166,8 @@ static int dfl_spi_altera_probe(struct dfl_device *dfl_dev)
 
 	if (dfl_dev->revision == FME_FEATURE_REV_MAX10_SPI_N5010)
 		strscpy(board_info.modalias, "m10-n5010", SPI_NAME_SIZE);
+	else if (dfl_dev->revision == FME_FEATURE_REV_MAX10_SPI_N5014)
+		strscpy(board_info.modalias, "m10-n5014", SPI_NAME_SIZE);
 	else
 		strscpy(board_info.modalias, "m10-d5005", SPI_NAME_SIZE);
 
